cliquet/storage/generators.py

```diff
@@ -11,7 +11,7 @@ class Generator(object):
     resource level to validate record id in requests paths.
     """
 
-    regexp = r'^[a-zA-Z0-9\-]+$'
+    regexp = r'^[a-zA-Z0-9][a-zA-Z0-9_-]*$'
```

Comment:
Yes I am sure we don't need backslash because the dash is at the end of the list.
I will add a test.